presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1832,4 +1862,70 @@ void PrestoServer::registerTraceNodeFactories() {
         return nullptr;
       });
 }
+
+void PrestoServer::optimizeExpressions(
```

Comment:
Lets move this code from PrestoServer into a file in presto_cpp/main/types. But do all the http request part of the processing here and the rest of the logic is in the inner function that uses more basic and Velox types.